src/jrd/extds/IscDS.cpp

```diff
@@ -144,17 +144,20 @@ void IscConnection::attach(thread_db* tdbb)
 		}
 	}
 
-	char buff[16];
+	char buff[64];
 	{
 		EngineCallbackGuard guard(tdbb, *this, FB_FUNCTION);
 
-		const char info[] = {isc_info_db_sql_dialect, isc_info_end};
+		const unsigned char info[] = {isc_info_db_sql_dialect, fb_info_provider_features, isc_info_end};
 		m_iscProvider.isc_database_info(&status, &m_handle, sizeof(info), info, sizeof(buff), buff);
 	}
 	if (status->getState() & IStatus::STATE_ERRORS) {
 		raise(&status, tdbb, "isc_database_info");
 	}
 
+	memset(m_features, false, sizeof(m_features));
+    m_sqlDialect = 1;
+
```

Comment:
Some extra whitespace seems to exist before m_sqlDialect.

Reply:
That's strange, I was sure that IDE is configured to have 4 spaces per tab...